rfcs/021-data_science_in_the_pipeline/README.md

```diff
@@ -23,27 +23,26 @@ We can write down some desiderata for any proposed solution:
 - DS models are persistent and separate to DS inferrers
 - DS models can be retrained on demand
 
-These suggest 3 distinct, but loosely coupled, services:
+These suggest 3 types of distinct, but loosely coupled, services:
 
-- **Inferrer**: A Python service that provides a synchronous API (most likely RESTful HTTP) that consumes whatever is needed to infer the new data, which it outputs.
-- **Model Trainer**: A Python service that can consume records from the catalogue index in bulk in order to train a model, and outputs/stores a persistent representation of this model for the *inferrer* to use.
-- **Inference Manager**: A Scala service that lives in the existing pipeline and contains "the usual" Wellcome message-passing, Akka, etc logic & libraries, which sychronously calls the `inferrer` and attaches the new data to the work/image (by populating a field) before passing it along.
+- **Inferrer**: A Python service that provides a synchronous API (most likely RESTful HTTP) that consumes whatever is needed to infer the new data, which it outputs. There can be multiple different inferrers.
+- **Inference Manager**: A Scala service that lives in the existing pipeline and contains "the usual" Wellcome message-passing, Akka, etc logic & libraries. It performs any work that is required by all of the inferrers, synchronously calls all of them and attaches the new data from them to the work/image (by populating a field) before passing it along. There is one inference manager for all the inferrers.
+- **Model Trainer**: A Python service that can consume records from the catalogue index in bulk in order to train a model, and outputs/stores a persistent representation of this model for an *inferrer* to use. There is **optionally** one model trainer for each inferrer.
 
 The usage of these services would look like this:
 
-![An architecture diagram for data science services](https://user-images.githubusercontent.com/4429247/76087593-631cc700-5fae-11ea-819f-3545e6dc7c65.png)
+![An architecture diagram for data science services](https://user-images.githubusercontent.com/4429247/88694368-5bde8d00-d0f8-11ea-8d0b-4b1c7687877b.png)
```

Comment:
trained models in EBS, not S3?

Reply:
Can't really work as the EBS volumes will be one-per-host (by definition, not by this design) and will scale up/down with the number of hosts. Also the trainer is still currently Fargate. I will clarify that the EBS volumes aren't intended to be persistent.

Reply:
Actually scratch that, they'll persist but not be attached to anything - but that doesn't change that EBS won't really work / isn't appropriate for model storage.